Sources/XMTPiOS/SigningKey.swift

```diff
@@ -19,6 +19,9 @@ import LibXMTP
 public protocol SigningKey {
 	/// A wallet address for this key
 	var address: String { get }
+	
+	/// Chain rpc url for the smart contract wallet
+	var chainRPCUrl: String? { get }
```

Comment:
@neekolas I think I need some knowledge sharing on how this chainRPC is supposed to be set. If it's okay for it to be on the signing key which is an interface over written by the integrator.

Reply:
This was all done a while ago, and not by me, so my recollection is a little fuzzy on where and when we need to pass in the chainRPC.

But generally it's a thing that dapps will have (it's the URL you'd get from an Infura or Alchemy dashboard). It wouldn't be "key specific", so it's probably safe to make it a client create parameter and save it there. We'd use the same RPC url for all signing keys.

We need it at the time we are creating or verifying smart contract wallet signatures.